kartothek/core/factory.py

```diff
@@ -99,20 +106,23 @@ def _instantiate_metadata_cache(self):
                 # backwards compat
                 self._cache_metadata = self._ds_callable()
             else:
-                self._cache_metadata = DatasetMetadata.load_from_store(
-                    uuid=self.dataset_uuid,
-                    store=self.store,
-                    load_schema=self.load_schema,
-                    load_all_indices=self.load_all_indices_flag,
-                )
+                self._exists = DatasetMetadata.exists(self.dataset_uuid, self.store)
+                if self._exists:
+                    self._cache_metadata = DatasetMetadata.load_from_store(
+                        uuid=self.dataset_uuid,
+                        store=self.store,
+                        load_schema=self.load_schema,
+                        load_all_indices=self.load_all_indices_flag,
+                    )
             if not self.load_dataset_metadata:
                 self._cache_metadata.metadata = {}
         self.is_loaded = True
         return self
 
     @property
     def dataset_metadata(self):
-        self._instantiate_metadata_cache()
+        if self._exists is not False:
```

Comment:
What happens w/o this existence check?

Reply:
I removed all unnecessary existents checks in the factory. Behavior should be now easier to understand, I hope.